tests/test_data/theglutenfreeaustrian.com/theglutenfreeaustrian_2.json

```diff
@@ -0,0 +1,79 @@
+{
+  "author": "Daniela Weiner",
+  "canonical_url": "https://theglutenfreeaustrian.com/gluten-free-brown-butter-snickerdoodles/",
+  "site_name": "The Gluten Free Austrian",
+  "host": "theglutenfreeaustrian.com",
+  "language": "en-US",
+  "title": "Gluten Free Brown Butter Snickerdoodles",
+  "ingredients": [
+    "140 grams unsalted butter (which we will brown)",
+    "90 grams granulated sugar",
+    "50 grams light brown sugar",
+    "1 large egg, at room temperature",
+    "1 teaspoon vanilla extract",
+    "195 grams gluten free multi purpose flour",
+    "10 grams cornstarch",
+    "1 teaspoon cream of tartar",
+    "1/2 teaspoon baking soda",
+    "1/2 teaspoon kosher salt",
+    "50 g granulated sugar",
+    "1 teaspoon ground cinnamon"
+  ],
+  "ingredient_groups": [
+    {
+      "ingredients": [
+        "140 grams unsalted butter (which we will brown)",
+        "90 grams granulated sugar",
+        "50 grams light brown sugar",
+        "1 large egg, at room temperature",
+        "1 teaspoon vanilla extract",
+        "195 grams gluten free multi purpose flour",
+        "10 grams cornstarch",
+        "1 teaspoon cream of tartar",
+        "1/2 teaspoon baking soda",
+        "1/2 teaspoon kosher salt"
+      ],
+      "purpose": "Gluten Free Brown Butter Snickerdoodles"
```

Comment:
What is better in general, I wonder:

  * To return `null` as the ingredient group instead of the recipe title when they are the same...
  * ...or to return the `title` instead of null as the purpose when we can't find a grouping name in the source?

Reply:
>To return null as the ingredient group instead of the recipe title when they are the same...

I think we should continue to return `null` as the purpose & in place of the title would be a nice cleanup.

Personally I use the `ingredient_groups` in other projects where I rely on the presence of all `null` purposes as a functional trigger. Of course this could be rewritten to fit a new setup but I think clearer to return a `null` value when there is no purpose.

Reply:
Do you see this change being implemented on a case by case basis or applied directly the ingredient_groups functionality?

Reply:
Case-by-case is probably best to begin with; if we find later that we're repeating nearly-identical code across lots of scrapers, then we could refactor; but I think it's the kind of thing that is likely to vary based on the way a site chooses to display their recipe info.

Reply:
I don't imagine this will be a common issue to run into but who knows!